### ISSUE TITLE
Add DefaultRectangular iterators that iterate in CMO

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#11377]
With CMO arrays being added in #11316, it would be good to be able to iterate through them in their natural order.  #11316 includes some commented out 'these_help' iterators that do this, but enabling them by default for CMO arrays causes problems for tests that expected RMO, for example:

```
fillRandom(CMOarray);
```

Fills the array in CMO, so the expected "random" values go down the columns first while tests expect them to go across the rows first.

In order to enable iterators that walk in the array's natural order by default, we'll need to decide what to do with the tests that have this sort of problem.

**Comment by bradcray** [comment]
I don't think that storing an array in CMO order should affect its default serial iteration order because then we lose the property that Chapel computations need not change when domain maps change, which is an important property for us.  I could imagine adding a "naturalIterOrder()" which could be used when you didn't care about the iteration order.  And clearly forall loops could use an arbitrary iteration order since they're unordered to begin with.

**Comment by daviditen** [comment]
> I don't think that storing an array in CMO order should affect its default serial iteration order because then we lose the property that Chapel computations need not change when domain maps change, which is an important property for us.

OK, that seems fair enough.

> I could imagine adding a "naturalIterOrder()" which could be used when you didn't care about the iteration order. And clearly forall loops could use an arbitrary iteration order since they're unordered to begin with.

Then this issue becomes: "Rename the currently commented out CMO iterators so they aren't called by default and make sure parallel versions are working."  Works for me.
